src/isEqual.test.ts

```diff
@@ -0,0 +1,47 @@
+import { isEqual } from './isEqual';
+
+test('data-first', () => {
+  expect(isEqual(2, 2)).toBe(true);
+  expect(isEqual('hello', 'hello')).toBe(true);
+  expect(isEqual(NaN, NaN)).toBe(true);
+  expect(isEqual(null, null)).toBe(true);
+  expect(isEqual(undefined, undefined)).toBe(true);
+
+  const symbol = Symbol('hello');
+  expect(isEqual(symbol, symbol)).toBe(true);
+  // @ts-expect-error these types don't overlap
+  expect(isEqual(symbol, Symbol('hello'))).toBe(false);
+
+  const obj = {};
+  expect(isEqual(obj, obj)).toBe(true);
+  expect(isEqual(obj, {})).toBe(false);
+
+  expect(isEqual(2, 3)).toBe(false);
+  // @ts-expect-error these types don't overlap
+  expect(isEqual(2 as 2 | 4, 3)).toBe(false);
```

Comment:
Split these into separate tests under a `describe` block so that each test passes or fails on it's own, and has a name that describes it's semantics.